router/email-status.go

```diff
@@ -34,4 +34,38 @@ func NewEmailStatusEngine(
 
 		ctx.JSON(http.StatusOK, res)
 	})
+
+	r.PUT(cfg.UpdateEmailStatus, func(ctx *gin.Context) {
+		utils.Logger.Info("Received updateEmailStatus request")
+
+		id := ctx.Param("id")
+		var payload mailer.EmailStatus
+		if err := ctx.ShouldBindJSON(&payload); err != nil {
+			ctx.JSON(http.StatusBadRequest, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+
+		if payload.Status == "" {
+			ctx.JSON(http.StatusBadRequest, gin.H{
+				"error": "status field cannot be empty",
+			})
+			return
+		}
+
```

Comment:
I think in addition to validating that the status cannot be empty, we should also validate the email status against a predefined enum. We want to avoid passing strings freely to represent the email status.

Reply:
Nice advise, but where is the enum? I can't find it

Reply:
or is it not implemented yet? If it's not implemented yet, do you have suggestion on what are the list of status for the enum?

Reply:
I couldn’t find the enum either. Currently, `success` and `failed` are already implemented for tracking the status when an email is sent. However, I think we also need additional statuses that can be manually edited to reflect the progress of the communication, such as `in_progress` and `completed`.

Here’s a proposed list of statuses:
- success: The email was successfully sent to the vendor.
- failed: The email failed to send due to an error.
- in_progress: The communication with the vendor is ongoing.
- completed: The communication with the vendor has been finalized.

What do you think of this approach?

Reply:
I dont really understand the need for "completed"

Reply:
> I dont really understand the need for "completed"
up

Reply:
Here's my understanding:
The default status would be either failed/success. If it's success, then If the vendor replied the email, then the user should set it to "in progress" because the communication is now active and ongoing. When the deal is sealed and the conversation is closed, the the user should set it to "completed".

Reply:
I believe the initial intention of this feature is to track whether the email that was sent was successful or failed, not the status of the conversation itself. I think it belongs to a different domain because "failed" or "successful" would then refer to whether the conversation or negotiation failed or succeeded